api/v1beta1/bgppeer_webhook.go

```diff
@@ -168,14 +166,6 @@ func (bgpPeer *BGPPeer) validateBGPPeerConfig(existingBGPPeersList *BGPPeerList)
 		return field.Invalid(field.NewPath("spec").Child("SrcAddress"), srcAddr,
 			fmt.Sprintf("Invalid BGPPeer source address %s", srcAddr))
 	}
-
-	for _, BGPPeer := range existingBGPPeersList.Items {
-		if remoteASN == BGPPeer.Spec.ASN && address == BGPPeer.Spec.Address && myASN == BGPPeer.Spec.MyASN {
```

Comment:
Can't we check if the name is different from the current peer instead of removing the check?

Reply:
during update it will have the same peer name , as I said in the PR I don't want block the update of BGPPeer CR because of this false positive if u have a way to tell its an iupdate that will help I was trying deepequal check but during update deepequal will fail so I might incorrectly run the check

Reply:
so if the name match one of existing skip the check ? ok what if someone edit an existing peer and made it match an existing session that will slip through , correct ?

Reply:
I am suggesting changing
```
 		if remoteASN == BGPPeer.Spec.ASN && address == BGPPeer.Spec.Address && myASN == BGPPeer.Spec.MyASN { 
```
to 
```
 		if remoteASN == BGPPeer.Spec.ASN && address == BGPPeer.Spec.Address && myASN == BGPPeer.Spec.MyASN && remoteName != BGPPeer.Name {
```
So we only check if we have a different peer with the same values. It will filter only if we are comparing the same pr.
I think that would work.

Reply:
I think this name check needed everywhere where I comparre against extsing list to not block update ?

Reply:
Yeah, my bad I did not notice while reviewing. We could add a test for that (at least for updating the bgppeer)